securedrop_client/gui/widgets.py

```diff
@@ -977,6 +985,44 @@ def update(self, sources: List[Source]) -> List[str]:
 
         return deleted_uuids
 
+    def initial_update(self, sources: List[Source]):
+        """
+        Initialise the list with the passed in list of sources.
+        """
+        sources.reverse()
```

Comment:
you can avoid having to reverse the sources list by inserting new widgets to the top of the SourceList in your add_source method